tests/test_optimizely.py

```diff
@@ -1173,25 +1173,28 @@ def test_get_variation__invalid_object(self):
 
     mock_logging.assert_called_once_with(enums.LogLevels.ERROR, 'Datafile has invalid format. Failing "get_variation".')
 
-  def test_is_feature_enabled__returns_false_for_none_feature_key(self):
-    """ Test that is_feature_enabled returns false if the provided feature key is None. """
+  def test_is_feature_enabled__returns_false_for_none_or_empty_feature_key(self):
+    """ Test that is_feature_enabled returns false if the provided feature key is empty or None. """
 
     opt_obj = optimizely.Optimizely(json.dumps(self.config_dict_with_features))
 
     with mock.patch('optimizely.logger.NoOpLogger.log') as mock_logger:
       self.assertFalse(opt_obj.is_feature_enabled(None, 'test_user'))
+      self.assertFalse(opt_obj.is_feature_enabled('', 'test_user'))
+    mock_logger.assert_called_with(enums.LogLevels.ERROR, enums.Errors.NONE_FEATURE_KEY_PARAMETER)
+    self.assertEqual(2, mock_logger.call_count)
 
-    mock_logger.assert_called_once_with(enums.LogLevels.ERROR, enums.Errors.NONE_FEATURE_KEY_PARAMETER)
-
-  def test_is_feature_enabled__returns_false_for_none_user_id(self):
-    """ Test that is_feature_enabled returns false if the provided user ID is None. """
+  def test_is_feature_enabled__returns_false_for_none_or_empty_user_id(self):
+    """ Test that is_feature_enabled returns false if the provided user ID is empty or None. """
 
     opt_obj = optimizely.Optimizely(json.dumps(self.config_dict_with_features))
 
     with mock.patch('optimizely.logger.NoOpLogger.log') as mock_logger:
       self.assertFalse(opt_obj.is_feature_enabled('feature_key', None))
+      self.assertFalse(opt_obj.is_feature_enabled('feature_key', ''))
 
-    mock_logger.assert_called_once_with(enums.LogLevels.ERROR, enums.Errors.NONE_USER_ID_PARAMETER)
+    mock_logger.assert_called_with(enums.LogLevels.ERROR, enums.Errors.INVALID_USER_ID_ERROR)
```

Comment:
This check only asserts the last call and not both calls. What you can do is `assertFalse`and check log message and then again do the same thing.

Reply:
I think you have taken a similar approach below.